pvlib/temperature.py

```diff
@@ -978,3 +978,397 @@ def prilliman(temp_cell, wind_speed, unit_mass=11.1, coefficients=None):
+
+    du_wind : numeric
+        Influence of wind speed on combined heat transfer coefficient
+        [(W/m^2)/C/(m/s)]
+
+    module_efficiency : float
+        The electrical efficiency of the module. [-]
+
+    absorptance : float
+        The light absorptance of the module. [-]
+
+    Returns
+    -------
+    numeric, values in degrees C.
+
+    References
+    ----------
+    .. [1] A. Driesse et al, "PV Module Operating Temperature
+       Model Equivalence and Parameter Translation". 2022 IEEE
+       Photovoltaic Specialists Conference (PVSC), 2022.
+
+    See also
+    --------
+    pvlib.temperature.GenericLinearModel
+    """
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs, Sept. 2022
+
+    heat_input = poa_global * (absorptance - module_efficiency)
+    total_loss_factor = u_const + du_wind * wind_speed
+    temp_difference = heat_input / total_loss_factor
+
+    return temp_air + temp_difference
+
+
+class GenericLinearModel():
+    '''
+    A generic linear temperature model class that can both use and convert
+    parameters of several well-known equivalent or nearly equivalent models:
+    faiman, pvsyst, noct-sam and sapm.
+
+    Normally this class is used on an ad hoc basis to make a needed conversion
+    to a target model, and the converted parameters are then used for
+    subsequent simulations running the target model.  The parameters are
+    always returned as dictionaries that are compatible with the respective
+    model functions in pvlib.temperature.
+
+    An instance of the class represents a specific module type, therefore,
+    the module properties *efficiency* and *absorptance* must be provided.
+    Although some temperature models do not use these properties, they
+    nevertheless exist and affect operating temperature.
+
+    Since these are empirical models, their parameters are determined
+    from a limited set of measurements (usually high irradiance).
+    The values of efficiency and absorptance should be interpreted
+    as representative of the efficiency and absorptance actually
+    occuring during those measurements.
+
+    When used for time series simulation, efficiency and absorptance may
+    vary, and each empirical model handles this differently.
+
+    Parameters
+    ----------
+    module_efficiency : float
+        The electrical efficiency of the module. [-]
+
+    absorptance : float
+        The light absorptance of the module. [-]
+
+    Notes
+    -----
+    After creating a GenericLinearModel object using the module properties,
+    one of the use_* methods must be called to provide thermal model
+    parameters.
+
+    References
+    ----------
+    .. [1] A. Driesse et al, "PV Module Operating Temperature
+       Model Equivalence and Parameter Translation". 2022 IEEE
+       Photovoltaic Specialists Conference (PVSC), 2022.
+
+    Examples
+    --------
+    glm = GenericLinearModel(module_efficiency=0.19, absorptance=0.88)
```

Comment:
This section is currently rendering as plaintext but it will render as python code if you prepend `>>>` to input lines. See `sapm_cell` for an example.